src/reducers/shelf/spec.test.ts

```diff
@@ -364,6 +366,483 @@ describe('reducers/shelf/spec', () => {
+        ]
+      });
+    });
+
+    it('should correctly convert back to a specific function', () => {
+      const shelfSpec = shelfSpecReducer(
+        {
+          ...DEFAULT_SHELF_UNIT_SPEC,
+          encoding: {},
+          anyEncodings: [
+            {aggregate: {name: 'aggregate', enum: ['mean', 'median']}, bin: '?', channel: '?', field: 'a'}
+          ]
+        },
+        {
+          type: SHELF_FUNCTION_DISABLE_WILDCARD,
+          payload: {
+            shelfId: {channel: '?', index: 0},
+          }
+        },
+        schema
+      );
+
+      expect(shelfSpec).toEqual({
+        ...DEFAULT_SHELF_UNIT_SPEC,
+        anyEncodings: [
+          {bin: true, channel: '?', field: 'a'}
+        ]
+      });
+    });
+
+    it('should correctly convert back to a specific function', () => {
+      const shelfSpec = shelfSpecReducer(
+        {
+          ...DEFAULT_SHELF_UNIT_SPEC,
+          encoding: {
+            x: {aggregate: {name: 'aggregate', enum: ['min', 'mean', 'median']}, field: 'a', type: 'quantitative'}
+          }
+        },
+        {
+          type: SHELF_FUNCTION_DISABLE_WILDCARD,
+          payload: {
+            shelfId: {channel: 'x'},
+          }
+        },
+        schema
+      );
+
+      expect(shelfSpec).toEqual({
+        ...DEFAULT_SHELF_UNIT_SPEC,
+        encoding: {
+          x: {aggregate: 'min', field: 'a', type: 'quantitative'}
+        }
+      });
+    });
+
+    it('should correctly convert back to a specific function', () => {
+      const shelfSpec = shelfSpecReducer(
+        {
+          ...DEFAULT_SHELF_UNIT_SPEC,
+          encoding: {
+            x: {
+              aggregate: {name: 'aggregate', enum: ['mean', 'median']},
+              bin: '?',
+              timeUnit: {name: 'timeUnit', enum: ['year']},
+              field: 'a',
+              type: 'quantitative'
+            }
+          }
+        },
+        {
+          type: SHELF_FUNCTION_DISABLE_WILDCARD,
+          payload: {
+            shelfId: {channel: 'x'},
+          }
+        },
+        schema
+      );
+
+      expect(shelfSpec).toEqual({
+        ...DEFAULT_SHELF_UNIT_SPEC,
+        encoding: {
+          x: {bin: true, field: 'a', type: 'quantitative'}
+        }
+      });
+    });
+
+    it('should correctly convert back to a specific function when no wildcard functions were selected', () => {
+      const shelfSpec = shelfSpecReducer(
+        {
+          ...DEFAULT_SHELF_UNIT_SPEC,
+          encoding: {
+            x: {field: 'a', type: 'quantitative'}
+          }
+        },
+        {
+          type: SHELF_FUNCTION_DISABLE_WILDCARD,
+          payload: {
+            shelfId: {channel: 'x'},
+          }
+        },
+        schema
+      );
+
+      expect(shelfSpec).toEqual({
+        ...DEFAULT_SHELF_UNIT_SPEC,
+        encoding: {
+          x: {field: 'a', type: 'quantitative'}
+        }
+      });
+    });
+  });
+
+  describe(SHELF_FUNCTION_ENABLE_WILDCARD, () => {
```

Comment:
I would add a case where previously there is no function, and then you enable wildcard.  
How do we distinguish between (1) no function, wildcard disabled and (2) no function, wildcard enabled.